specification/monitor/resource-manager/Microsoft.Insights/preview/2023-06-01-preview/privateLinkScopes_API.json

```diff
@@ -0,0 +1,1110 @@
+        "privateEndpointConnectionName": {
+          "description": "The private endpoint connection name associated to the private endpoint on which we want to apply the specific access mode settings.",
+          "type": "string"
+        },
+        "queryAccessMode": {
+          "description": "Specifies the access mode of queries through the specified private endpoint connection in the exclusion.",
+          "$ref": "#/definitions/AccessMode"
+        },
+        "ingestionAccessMode": {
+          "description": "Specifies the access mode of ingestion through the specified private endpoint connection in the exclusion.",
+          "$ref": "#/definitions/AccessMode"
+        }
+      }
+    },
+    "AccessMode": {
+      "description": "Access mode types.",
+      "type": "string",
+      "enum": [
+        "Open",
+        "PrivateOnly"
+      ],
+      "x-ms-enum": {
+        "name": "AccessMode",
+        "modelAsString": true
+      }
+    },
+    "AzureMonitorPrivateLinkScopeListResult": {
+      "description": "Describes the list of Azure Monitor PrivateLinkScope resources.",
+      "type": "object",
+      "required": [
+        "value"
+      ],
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "List of Azure Monitor PrivateLinkScope definitions.",
+          "items": {
+            "$ref": "#/definitions/AzureMonitorPrivateLinkScope"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URI to get the next set of Azure Monitor PrivateLinkScope definitions if too many PrivateLinkScopes where returned in the result set."
+        }
+      }
+    },
+    "ScopedResourceListResult": {
+      "description": "A list of scoped resources in a private link scope.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Array of results.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ScopedResource"
+          },
+          "readOnly": true
+        },
+        "nextLink": {
+          "description": "Link to retrieve next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "ScopedResource": {
+      "description": "A private link scoped resource",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ScopedResourceProperties",
+          "description": "Resource properties.",
+          "x-ms-client-flatten": true
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData",
+          "description": "System data",
+          "readOnly": true
+        }
+      }
+    },
+    "ScopedResourceProperties": {
```

Comment:
>ScopedResourceProperties [](http://example.com/codeflow?start=5&length=24)

I notice that HybridCompute seems to implement a similar payload, but without the subscriptionLocation, or kind.

What 'kind' values are supposed to be allowed?

What kind of a subscription are we talking about here? An Azure Subscription?

Reply:
Answer as I understand it from chat: Its a common contract _by convention_. I.e. we'd like it to be consistent for the sake of API consistency. /cc @rkmanda